### PR TITLE
IDC: non-admin screen

### PR DESCRIPTION
**Pull request by sergeymitr**
#### Changes proposed in this Pull Request:
- create the non-admin IDC screen
- adjust Connection Manager UI package to be available for non-admins so they could see the IDC screen there

#### Jetpack product discussion
p9dueE-3pz-p2#comment-6479

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Rebuild the Connection Manager package: `jetpack build packages/connection-ui`.
2. Connect Jetpack, go to Jetpack Debug and activate the IDC Simulator.
3. Go to "Jetpack Debug -> IDC Simulator", enable "IDC Simulation" and enter a "Spoof URL".
4. Edit a post and save to make the site sync. Wait a few seconds.
5. Go to "Tools -> Connection Manager", confirm that the RNA IDC screen appears.
6. Log in as non-admin, go to "Tools -> Connection Manager". Confirm that the IDC screen looks like the attached screenshot, and the "Learn more..." link works.
Screen Shot 2021-12-08 at 9 57 21 AM

##### Storybook
7. Go to `projects/js-packages/storybook`, run `pnpm install` and `pnpm run storybook:dev`.
8. When the Storybook loads, navigate to "Identity Crisis -> Admin Screen", find the story "Non Admin". Confirm that it looks like the screenshot.

### CODE REVIEW

**Comment by kbrown9** [comment]
I've tested on a dev site and in storybook, and everything looks good.

Do we want to display the IDC UI to non-admins even after a admin has confirmed that the site should be kept in safe mode?

**Comment by sergeymitr** [comment]
Hi @kbrown9.

That's a good catch, thanks!
I dug a bit more into it, and it looks like the IDC UI does show up for non-admins in safe mode everywhere, including the Jetpack Dashboard.
It gets rendered, and flickers for a moment when the Dashboard is loading, but disappears quickly because Jetpack hides all notices, including that one: https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/_inc/client/components/admin-notices/index.jsx#L103-L110

I adjusted Connection Manager UI to match that behavior, with an exception that we don't hide the notices.
So non-admins still see the IDC UI, but the rest of the page still renders if an admin decided to stay in safe mode.

Connection Manager UI is supposed to serve as an usage example, so please take a look and let me know if you think that'll suffice: https://github.com/Automattic/jetpack/pull/22050/commits/69ce3dc10b00ccd35e6ff4193e3b4b47b8e16558

**Review by samiff — COMMENTED** [review]
Tested nicely for me 👍 

> So non-admins still see the IDC UI, but the rest of the page still renders if an admin decided to stay in safe mode.

This part worked.

File: projects/js-packages/idc/components/idc-screen/screen-non-admin.jsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+	return (
+		<React.Fragment>
+			<h2>{ customContent.nonAdminTitle || __( 'Safe Mode has been activated', 'jetpack' ) }</h2>
+
+			<p>
+				{ customContent.nonAdminBodyText ||
+					createInterpolateElement(
+						__(
+							'Your site is in Safe Mode because you have 2 Jetpack-powered sites that appear to be duplicates. ' +
```
[the remark this answers is not in the document]
Comment (sergeymitr) [reply projects/js-packages/idc/components/idc-screen/screen-non-admin.jsx]: Good catch, thanks 👍

**Review by kbrown9 — APPROVED** [review]
This looks good to me, and everything tested well.

**Review by kbrown9 — APPROVED** [review]
I'm reapproving after the merge of the origin/master branch, which only changed the `composer.lock` file.

**Comment by sergeymitr** [comment]
Thank you @kbrown9 🎉

**Comment by sergeymitr** [comment]
r236512-wpcom

**Pull request merged**